### DISCUSSION TITLE
disable/enable components

### DISCUSSION THREAD

**Discussion by Chasikanaft** [ArthurSonzogni/FTXUI#390] [Q&A] [answered]
How enable/disable components like a buttons?
or change properties like a "label"

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello!

To enable/disable an arbitrary component, you can wrap them with [`Maybe`](https://arthursonzogni.github.io/FTXUI/#component-maybe).
For example:
```cpp
bool button_enabled = true;
auto button = Button("Disable this button ", []{
  button_enabled = false;
});
auto maybe_button = Maybe(button, &button_enabled);
```

To change the button's state, you can pass a reference to its state and modify it:
```cpp
std::string label = "button label";
auto button = Button(&label, [&] {
  label = "new label";
});
```

**Reply by Chasikanaft** [reply]
what about animation? like button makes green when pressed?

**Reply by ArthurSonzogni** [reply]
Use`ButtonOption::Animated` and related functions overload.

**Reply by Chasikanaft** [reply]
or maybe exist simple way do "pressed" button?

**Reply by ArthurSonzogni** [reply]
Sorry, I don't understand the question.

**Reply by Chasikanaft** [reply]
Is it possible to make a push button? Relatively speaking, when a button is pressed, an animation is played that lasts until it is pressed again. I looked at the methods of `ButtonOption::Animated` and did not quite understand how to do it not through changing states according to the example - set the variable "`bool _preset`" depending on which changes when you click:
```
if (_preset) {
`**ButtonOption linked object**.Animated().animated_colors.background.Set(Color:: GrayDark, Color::Green, inf);
}
else
{
**ButtonOption linked object**.Animated().animated_colors.background.Set(Color:: Red, Color::GrayDark, inf);
}
```

**Reply by ArthurSonzogni** [reply]
It sounds like you are describing a `Checkbox`.
With `CheckboxOption::transform`, you can describe how to render it as a push button.

**Reply by Chasikanaft** [reply]
Can i ask you write some example? Please :)
